src/miner.cpp

```diff
@@ -308,7 +358,7 @@ void BlockAssembler::SortForBlock(const CTxMemPool::setEntries& package, std::ve
 // Each time through the loop, we compare the best transaction in
 // mapModifiedTxs with the next transaction in the mempool to decide what
 // transaction package to work on next.
-void BlockAssembler::addPackageTxs(int &nPackagesSelected, int &nDescendantsUpdated)
+void BlockAssembler::addPackageTxs(int& nPackagesSelected, int& nDescendantsUpdated, CFeeRate* min_package_fee_rate)
```

Comment:
`min_package_fee_rate` seems very cheap to calculate. Perhaps just make it a reference so it's always returned.

(even better would be:

```suggestion
std::tuple<int, int, CFeeRate> BlockAssembler::addPackageTxs()
```

but that's maybe beyond the scope of this PR)

Reply:
agree that updating `min_package_fee_rate` is pretty cheap, there's no additional calculation, its just updating the stored value, but might do so up to ~once per mempool transaction. 

but can you help me understand the context of why you're suggesting a reference? even if it's not a lot of calculation, seems unnecessary? 

(agree that returning a tuple instead of in/out params would be nice. but leaving other params as is within this PR.)